src/system/SystemPacketBuffer.cpp

```diff
@@ -140,7 +141,7 @@ void PacketBufferHandle::InternalRightSize()
     // Reallocate only if enough space will be saved.
     const uint8_t * const start   = mBuffer->ReserveStart();
     const uint8_t * const payload = mBuffer->Start();
-    const uint16_t usedSize       = static_cast<uint16_t>(payload - start + mBuffer->len);
+    const size_t usedSize         = static_cast<size_t>(static_cast<uint32_t>(payload - start) + mBuffer->len);
```

Comment:
Why the intermediate cast to uint32_t?  This doesn't make sense.

Reply:
This is because we are adding the difference to mBuffer->len which is a size_t and unsigned. payload - start is guaranteed to be positive, so this cast should be fine.

Reply:
None of that answers my question about why the intermediate cast is needed.  What happens if this is written as: `usedSize = (payload - start) + mBuffer->len;` without any casts?